docs/standard-library/algorithm-functions.md

```diff
@@ -2261,85 +2261,94 @@ otherwise the first element.
 
 ### Remarks
 
-The range referenced must be valid; all pointers must be dereferenceable and, within the sequence, the last position must be reachable from the first by incrementation.
-
-The complexity is linear with at most *`count`* operations.
+*`count`* must be non-negative, and there must be at least *`count`* elements in the range starting at *`first`*.
 
 ### Example
 
+This example depicts a user-defined function object class, production code often
+uses [`lambda`](https://docs.microsoft.com/en-us/cpp/cpp/lambda-expressions-in-cpp?view=msvc-170)s to achieve the same result with less code
+
+Example requires `c++17`
+
 ```cpp
 // alg_for_each_n.cpp
 // compile with: /EHsc
-#include <vector>
 #include <algorithm>
 #include <iostream>
+#include <vector>
+using namespace std;  // Global standard namespace (bad practice)
 
 // The function object multiplies an element by a Factor
-template <class Type>
-class MultValue
+template <class Type> class MultValue
 {
-private:
-    Type Factor;   // The value to multiply by
-public:
+  private:
+    Type Factor; // The value to multiply by
+  public:
     // Constructor initializes the value to multiply by
-    MultValue ( const Type& value ) : Factor ( value ) {
+    MultValue(const Type &value) : Factor(value)
+    {
     }
 
     // The function call for the element to be multiplied
-    void operator( ) ( Type& elem ) const
+    void operator()(Type &elem) const
     {
         elem *= Factor;
     }
 };
 
+template <class Vector> void print_vector(const Vector &vec)
+{
+    cout << "( ";
+
+    for (typename vector<int>::const_iterator iter = vec.begin(); iter != vec.end(); iter++)
```

Comment:
suggest using auto to hide the iterator type complexity, which isn't germane to the example